### ISSUE TITLE
New parser request: os-prober

### ISSUE THREAD

**Issue by don-rumata** [kellyjonbrazil/jc#316]
Hi, can you add support for `os-prober` output?

**Comment by kellyjonbrazil** [comment]
Hi - thank you for the parser suggestion! Can you provide some os-prober output samples? I haven't used this command before and I don't see much information about it.

**Comment by don-rumata** [comment]
My computer (win7+ubuntu):
`/dev/sda1:Windows 7 (loader):Windows:chain`

My VM with win10 only:
`/dev/sda1:Windows 10:Windows:chain`

My VM with win10 only (UEFI):
`/dev/sda1@/efi/Microsoft/Boot/bootmgfw.efi:Windows Boot Manager:Windows:efi`

Doc: https://github.com/campadrenalin/os-prober/blob/master/README

**Comment by kellyjonbrazil** [comment]
Thanks! It looks pretty straightforward to parse, so I can probably add that in the next release.

**Comment by kellyjonbrazil** [comment]
I added a working version of the `os-prober` parser in the `dev` branch. The output looks like this:
```
$ os-prober | jc --os-prober -p
    {
      "partition": "/dev/sda1",
      "name": "Windows 10",
      "short_name": "Windows",
      "type": "chain"
    }
```
If you'd like to test you can copy the parser file to your plugin directory. This parser requires `jc` v1.22.1 to run:

https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/os_prober.py

- Linux/unix: `$HOME/.local/share/jc/jcparsers`
- macOS: `$HOME/Library/Application Support/jc/jcparsers`
- Windows: `$LOCALAPPDATA\jc\jc\jcparsers`

**Comment by don-rumata** [comment]
My VM with win7 legacy:

```json
{
  "partition": "/dev/sda1",
  "name": "Windows 7",
  "short_name": "Windows",
  "type": "chain"
}
```
My VM with win10 legacy:

```json
{
  "partition": "/dev/sda1",
  "name": "Windows 10",
  "short_name": "Windows",
  "type": "chain"
}
```
My VM with win10 UEFI:

```json
{
  "partition": "/dev/sda1@/efi/Microsoft/Boot/bootmgfw.efi",
  "name": "Windows Boot Manager",
  "short_name": "Windows",
  "type": "efi"
}
```

Can you add a separator for the "@" symbol?

**Comment by kellyjonbrazil** [comment]
I could make a `partition_split` field that splits on the `@` character:
```
{
  "partition": "/dev/sda1@/efi/Microsoft/Boot/bootmgfw.efi",
  "partition_split": [
    "/dev/sda1",
    "/efi/Microsoft/Boot/bootmgfw.efi"
  ],
  "name": "Windows Boot Manager",
  "short_name": "Windows",
  "type": "efi"
}
```
Are there any other split characters that you know of?

**Comment by don-rumata** [comment]
https://github.com/campadrenalin/os-prober/blob/7ea6022b1071818681f8b1e5afca1ee758923ee6/src/probes/os/mounted/x86/05efi#L66
Apparently this is an efi-specific separator

```
cd os-prober/src/probes/os/mounted/x86
grep result ./*
```
```
./05efi:                        result "${partition}@/$efi/${entry}:efi"
./10freedos:    result "$1:FreeDOS:$label:chain"
./10qnx:        result "$partition:QNX:$label:chain"
./20microsoft:result "${partition}:${long}:${label}:chain"
./30utility:result "${partition}:${long}:${label}:chain"
./70hurd:       result "$partition:GNU/Hurd:$label:hurd"
./80minix:      result "$partition:Minix:$label:$boot"
./83haiku:      result "$partition:Haiku:$label:chain"
./90solaris:    result "$partition:Solaris/IA32:$label:chain"
```

**Comment by don-rumata** [comment]
Can you do that?

```json
"partition": "/dev/sda1",
"efi_bootmgr": "/efi/Microsoft/Boot/bootmgfw.efi"
```

**Comment by kellyjonbrazil** [comment]
I have this working in the `dev` branch now:
https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/os_prober.py
```
$ echo '/dev/sda1@/efi/Microsoft/Boot/bootmgfw.efi:Windows Boot Manager:Windows:efi' | jc --os-prober -p
{
  "partition": "/dev/sda1",
  "name": "Windows Boot Manager",
  "short_name": "Windows",
  "type": "efi",
  "efi_bootmgr": "/efi/Microsoft/Boot/bootmgfw.efi"
}
```
The `-r` or `--raw` option doesn't split the partition (old behavior):
```
$ echo '/dev/sda1@/efi/Microsoft/Boot/bootmgfw.efi:Windows Boot Manager:Windows:efi' | jc --os-prober -pr
{
  "partition": "/dev/sda1@/efi/Microsoft/Boot/bootmgfw.efi",
  "name": "Windows Boot Manager",
  "short_name": "Windows",
  "type": "efi"
}
```
If there are no issues this will be included in the next release (v1.22.3)
